File: src/test/java/ar/edu/unq/unidad1/wop/dao/JDBCPersonajeDAOTest.java
package ar.edu.unq.unidad1.wop.dao;

import ar.edu.unq.unidad1.wop.dao.impl.JDBCPersonajeDAO;
import ar.edu.unq.unidad1.wop.modelo.Item;
import ar.edu.unq.unidad1.wop.modelo.Personaje;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.HashSet;

import static java.util.logging.LogManager.getLogManager;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class JDBCPersonajeDAOTest {
    private PersonajeDAO dao = new JDBCPersonajeDAO();
    private Personaje maguito;

//    Descomentar para ver logs.

//    @BeforeAll
//    static void setupLogging() {
//        try {
//            // Cargamos la configuración de logging
//            getLogManager().readConfiguration(
//                JDBCPersonajeDAOTest.class.getClassLoader().getResourceAsStream("logging.properties")
//            );
//        } catch (Exception e) {
//            System.err.println("No se pudo cargar la configuración de logging: " + e.getMessage());
//        }
//    }

    @BeforeEach
    void crearModelo() {
        maguito = new Personaje("Maguito",
                15,
                2500,
                198,
                new HashSet<>()
                );
        maguito.recoger(new Item("Tunica gris", 1));
        maguito.recoger(new Item("Baculo gris", 5));
    }

    @Test
    void alGuardarYLuegoRecuperarSeObtieneObjetosSimilares() {
        dao.guardar(maguito);

        //Los personajes son iguales
        var otroMaguito = dao.recuperar("Maguito");
        assertEquals(maguito.getNombre(), otroMaguito.getNombre());
        assertEquals(maguito.getPesoMaximo(), otroMaguito.getPesoMaximo());
        assertEquals(maguito.getVida(), otroMaguito.getVida());
        assertEquals(maguito.getXp(), otroMaguito.getXp());
//        assertEquals(this.maguito.getInventario().size(), otroMaguito.getInventario().size());

        //Pero no son el mismo objeto =(
        //A esto nos referimos con "perdida de identidad"
        Assertions.assertNotEquals(maguito, otroMaguito);
    }

    @AfterEach
    void emilinarModelo() {
        dao.eliminar(maguito);
    }
}

File: src/test/java/ar/edu/unq/unidad1/wop/dao/JDBCPersonajeDAOTest.kt
package ar.edu.unq.unidad1.wop.dao

import ar.edu.unq.unidad1.wop.dao.impl.JDBCPersonajeDAO
import ar.edu.unq.unidad1.wop.modelo.Item
import ar.edu.unq.unidad1.wop.modelo.Personaje
import org.junit.jupiter.api.*
import org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS

@TestInstance(PER_CLASS)
class JDBCPersonajeDAOTest {
    private val dao: PersonajeDAO = JDBCPersonajeDAO()
    lateinit var maguito: Personaje

    @BeforeEach
    fun crearModelo() {
        maguito = Personaje("Maguito")
        maguito.pesoMaximo = 15
        maguito.vida = 198
        maguito.xp = 2500
        maguito.recoger(Item("Tunica gris", 1))
        maguito.recoger(Item("Baculo gris", 5))
    }

    @Test
    fun alGuardarYLuegoRecuperarSeObtieneObjetosSimilares() {
        dao.guardar(maguito)

        //Los personajes son iguales
        val otroMaguito = dao.recuperar("Maguito")
        Assertions.assertEquals(maguito.nombre, otroMaguito.nombre)
        Assertions.assertEquals(maguito.pesoMaximo.toLong(), otroMaguito.pesoMaximo.toLong())
        Assertions.assertEquals(maguito.vida.toLong(), otroMaguito.vida.toLong())
        Assertions.assertEquals(maguito.xp.toLong(), otroMaguito.xp.toLong())
        //		assertEquals(this.maguito.getInventario().size(), otroMaguito.getInventario().size());

        //Pero no son el mismo objeto =(
        //A esto nos referimos con "perdida de identidad"
        Assertions.assertTrue(maguito !== otroMaguito)
    }


    @AfterEach
    fun emilinarModelo() {
        dao.eliminar(maguito)
    }
}